server/routes/chat_routes.test.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ResponseObject } from '@hapi/hapi';
+import { Boom } from '@hapi/boom';
+
+import { Router } from '../../../../src/core/server/http/router';
+import { enhanceWithContext, triggerHandler } from './router.mock';
+import { httpServerMock } from '../../../../src/core/server/http/http_server.mocks';
+import { mockAgentFrameworkStorageService } from '../services/storage/agent_framework_storage_service.mock';
+import { mockOllyChatService } from '../services/chat/olly_chat_service.mock';
+import { loggerMock } from '../../../../src/core/server/logging/logger.mock';
+import { registerChatRoutes, GetSessionsSchema } from './chat_routes';
+import { ASSISTANT_API } from '../../common/constants/llm';
+
+const mockedLogger = loggerMock.create();
+
+const mockStorageGetSession = () => {
+  const getSessionResultMock = {
+    messages: [{ type: 'input' as const, contentType: 'text' as const, content: 'foo' }],
+    title: 'foo',
+    interactions: [
+      {
+        input: 'foo',
+        response: 'bar',
+        conversation_id: 'conversation-1',
+        interaction_id: 'interaction-1',
+        create_time: '',
+      },
+    ],
+    createdTimeMs: 0,
+    updatedTimeMs: 0,
+  };
+  const getSession = mockAgentFrameworkStorageService.getSession.mockReturnValueOnce(
+    Promise.resolve(getSessionResultMock)
+  );
+  return {
+    getSession,
+    getSessionResultMock,
+  };
+};
+
+const router = new Router(
+  '',
+  mockedLogger,
+  enhanceWithContext({
+    assistant_plugin: {
+      logger: mockedLogger,
+    },
+  })
+);
+registerChatRoutes(router, {
+  messageParsers: [],
+});
+
+const triggerGetSession = (sessionId: string) =>
+  triggerHandler(router, {
+    method: 'get',
+    path: `${ASSISTANT_API.SESSION}/{sessionId}`,
+    req: httpServerMock.createRawRequest({ params: { sessionId } }),
+  });
+const triggerGetSessions = (query: GetSessionsSchema) =>
+  triggerHandler(router, {
+    method: 'get',
+    path: ASSISTANT_API.SESSIONS,
+    req: httpServerMock.createRawRequest({ query }),
+  });
```

Comment:
Actually there are get_session.test.ts and get_sessions.test.ts files which have already done the unit test. Maybe remove the standalone files to keep just one test case?

Reply:
Updated